### PR TITLE
[acs-notifiers] introduce integration

### PR DESCRIPTION
**Pull request by maorfr**
part of https://issues.redhat.com/browse/APPSRE-10043

depends on https://github.com/app-sre/qontract-schemas/pull/625

design document: https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/102231

this integration iterates over ACS policies, extracts their Escalation Policy, and uses that to compose a Jira Notifier.

with the introduction of an Escalation Policy, in a follow up PR, we will also remove usage of hard coded (manually managed) `notifiers`. we will use the notifier type (`jira`) and name of the Escalation Policy instead.

this means that the name of the Notifier is the point of integration between `acs-policies` and `acs-notifiers` (related to https://github.com/app-sre/qontract-schemas/pull/604)

API docs: https://acs-cl4gmo2o5v7imos30r30.acs.rhcloud.com/main/apidocs#tag/NotifierService

### CODE REVIEW

**Comment by rporres** [comment]
A design doc would help understanding what's going on here :)

File: reconcile/acs_notifiers.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+            logging.info(f"Delete Jira notifier: {d.name}")
+            if not dry_run:
+                acs_api.delete_jira_notifier(d)
+
+    def run(
+        self,
+        dry_run: bool,
+    ) -> None:
+        vault_settings = get_app_interface_vault_settings()
+        secret_reader = create_secret_reader(use_vault=vault_settings.vault)
```
Comment (chassing) [inline reconcile/acs_notifiers.py:142]: not needed, just use `self.secret_reader`

File: reconcile/acs_notifiers.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            url=jira_board.server.server_url,
+            issue_type=jira_board.issue_type or "Task",
+            severity_priority_mappings=sorted(
+                [
+                    SeverityPriorityMapping(**vars(sp))
+                    for sp in jira_board.severity_priority_mappings.mappings
+                ],
+                key=lambda m: m.severity,
+            ),
+            custom_fields=custom_fields if custom_fields else {},
```
Comment (chassing) [inline reconcile/acs_notifiers.py:86]:
```suggestion
            custom_fields=custom_fields,
```
should be enough

File: reconcile/acs_notifiers.py
Comment (chassing) [inline reconcile/acs_notifiers.py:1]: Some unit tests would be great 🙊

File: reconcile/utils/acs/notifiers.py
Comment (chassing) [inline reconcile/utils/acs/notifiers.py:1]: What about some unit tests 🙈 ?

File: reconcile/acs_notifiers.py (hunk 1 above)
Comment (maorfr) [reply reconcile/acs_notifiers.py]: custom_fields=custom_fields if custom_fields else {}

File: reconcile/acs_notifiers.py (hunk 2 above)
Comment (maorfr) [reply reconcile/acs_notifiers.py]: https://github.com/app-sre/qontract-reconcile/pull/4246/commits/3785a1894470470b14165eea38537f9a4c5440f4

File: reconcile/acs_notifiers.py (hunk 1 above)
Comment (maorfr) [reply reconcile/acs_notifiers.py]: did more of the same for existing integrations in #4263

File: reconcile/test/test_acs_notifiers.py
Comment (chassing) [inline reconcile/test/test_acs_notifiers.py:1]: ❤️

**Review by chassing — APPROVED** [review]

**Pull request merged**